keep/workflowmanager/workflowscheduler.py

```diff
@@ -490,61 +487,53 @@ def _handle_event_workflows(self):
                             "tenant_id": tenant_id,
                         },
                     )
-                    self._finish_workflow_execution(
+                    await self._finish_workflow_execution(
                         tenant_id=tenant_id,
                         workflow_id=workflow_id,
                         workflow_execution_id=workflow_execution_id,
                         status=WorkflowStatus.ERROR,
                         error=f"Error getting alert by id: {e}",
                     )
                     continue
-            # Last, run the workflow
-            thread = threading.Thread(
-                target=self._run_workflow,
-                args=[tenant_id, workflow_id, workflow, workflow_execution_id, event],
-            )
-            thread.start()
-            self.threads.append(thread)
+            tasks.append(asyncio.create_task(self._run_workflow(tenant_id, workflow_id, workflow, workflow_execution_id, event)))
+            await asyncio.gather(*tasks)
```

Comment:
Again, not optimal to await for them here, but that's for the next iteration. For now, want to keep the code simple.